.github/workflows/unix-build.yml

```
@@ -0,0 +1,36 @@
+# Various unix builds
+name: unix CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    
+    defaults:
+      run:
+        shell: bash {0}
+
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Mac install dependencies
+      if: ${{ matrix.os == 'macos-latest' }}
```

I'd suggest [`runner.os`](https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context) instead. Its value is fixed, whereas next month you might want the matrix entry to be `macos-14` or so.

```suggestion
      if: runner.os == 'macOS'
```

(IIRC `==` is case insensitive anyway, but might as well use the documented capitalisation.)

---

Thanks.  At the moment this is very exploratory.  Partly an attempt to avoid AppVeyor as it's getting slower and slower to launch Windows MinGW builds, and partly to offload some of the workload with give to cirrus-ci as it has a maximum amount of CPU we can use, has a reduced number of concurrent jobs, and we're low on priority in the free tier.  (Or we get our bill payers to consider such tooling as something to pay for instead of just the usual corporate-ware.)

Current wrestle: submodules and gitattributes.  Shrug.  Should work!  Will beat it around the head some more :)